rdkit/Chem/AllChem.py

```diff
@@ -464,47 +465,170 @@ def AssignBondOrdersFromTemplate(refmol, mol):
   mol2 = rdchem.Mol(mol)
   # do the molecules match already?
   matching = mol2.GetSubstructMatch(refmol2)
-  if not matching: # no, they don't match
+  if not matching or mol2.GetNumBonds() != refmol2.GetNumBonds(): # no, they don't match, or some bonds are bad
     # check if bonds of mol are SINGLE
     for b in mol2.GetBonds():
-      if b.GetBondType() != BondType.SINGLE:
-        b.SetBondType(BondType.SINGLE)
-        b.SetIsAromatic(False)
+      b.SetBondType(BondType.SINGLE)
+      b.SetIsAromatic(False)
     # set the bonds of mol to SINGLE
     for b in refmol2.GetBonds():
       b.SetBondType(BondType.SINGLE)
       b.SetIsAromatic(False)
     # set atom charges to zero;
     for a in refmol2.GetAtoms():
       a.SetFormalCharge(0)
+      a.SetIsAromatic(False)
     for a in mol2.GetAtoms():
       a.SetFormalCharge(0)
-      
+      a.SetIsAromatic(False)
+
     matching = mol2.GetSubstructMatches(refmol2, uniquify=False)
+    # an inveerse match
+    if not matching:
+      inverse_matching = refmol2.GetSubstructMatches(mol2, uniquify=False)
+      if inverse_matching and len(inverse_matching[0]) == refmol2.GetNumAtoms():
+        matching = tuple(i[0] for i in sorted(enumerate(inverse_matching[0]),key=lambda x: x[1]))
+        matching = (matching,)
+
+    # add missing and remove excesive bonds
+    if len(GetMolFrags(mol2, sanitizeFrags=False)) > 1 or mol2.GetNumAtoms() != refmol2.GetNumAtoms():
+      fragments = zip(GetMolFrags(mol2, sanitizeFrags=False), GetMolFrags(mol2, asMols=True, sanitizeFrags=False))
+      fragments = sorted(fragments, key=lambda x: len(x), reverse=True) # sort by decreasing size
```

Comment:
This sort doesn't do anything because it's sorting on the length of the 2-tuples that make up fragments.
I think you mean:
`fragments = sorted(fragments, key=lambda x: len(x[0]), reverse=True)`